tests/firedrake/demos/test_demos_run.py

```diff
@@ -41,6 +42,7 @@
     Demo(("poisson", "poisson_mixed"), ["vtk"]),
     Demo(("quasigeostrophy_1layer", "qg_1layer_wave"), ["hypre", "vtk"]),
     Demo(("saddle_point_pc", "saddle_point_systems"), ["hypre", "mumps"]),
+    Demo(("time_dependent_rayleigh_benard", "timedep-rayleigh-benard"), ["mumps"])
```

Comment:
```suggestion
    Demo(("time_dependent_rayleigh_benard", "timedep-rayleigh-benard"), ["irksome", "mumps"])
```
We should probably be doing something like this. You will need to add something [here](https://github.com/firedrakeproject/firedrake/blob/master/tests/firedrake/conftest.py#L13) and also `pip install` Irksome inside `.github/workflows/core.yml`

Reply:
I can have a go at this.  So I'll assume this is your blessing to do so.

Reply:
We'll see if this makes the CI happy now.